core/server/common/src/main/java/alluxio/cli/validation/Utils.java

```diff
@@ -91,5 +93,75 @@ public static boolean isAlluxioRunning(String className) {
     return nodes;
   }
 
+  /**
+   * Executes a command in another process and check for its execution result.
+   *
+   * @param args array representation of the command to execute
+   * @return {@link ProcessExecutionResult} including the process's exit value, output and error
+   */
+  public static ProcessExecutionResult getResultFromProcess(String[] args) {
+    try {
+      Process process = Runtime.getRuntime().exec(args);
+      StringBuilder outputSb = new StringBuilder();
+      try (BufferedReader processOutputReader = new BufferedReader(
+          new InputStreamReader(process.getInputStream()))) {
+        String line;
+        while ((line = processOutputReader.readLine()) != null) {
+          outputSb.append(line);
+          outputSb.append(LINE_SEPARATOR);
+        }
+      }
+      StringBuilder errorSb = new StringBuilder();
+      try (BufferedReader processErrorReader = new BufferedReader(
+          new InputStreamReader(process.getErrorStream()))) {
+        String line;
+        while ((line = processErrorReader.readLine()) != null) {
+          errorSb.append(line);
+          errorSb.append(LINE_SEPARATOR);
+        }
+      }
+      process.waitFor();
+      return new ProcessExecutionResult(process.exitValue(), outputSb.toString().trim(),
+          errorSb.toString().trim());
+    } catch (IOException e) {
+      System.err.println("Failed to execute command.");
```

Comment:
Still feel that `e.getMessage()` should be also added to the output message here. That would make it easier to debug IO issues than just showing generic "Failed to execute command" message/

Reply:
Done.